### ISSUE TITLE
Java Modules v1.3.x

### ISSUE THREAD

**Issue by nhojpatrick** [hamcrest/JavaHamcrest#204]
Hamcrest future

Java modules was massive change, rolling out a patch on 1.3 which supports modules I feel would be useful for downstream projects that build upon, use or incorporate hamcrest into their libraries.

I'm specifically talking about "Automatic-Module-Name" being added into the MANIFEST.MF.

**Comment by sf105** [comment]
It sounds useful. What do we need to do in this space?

**Comment by nhojpatrick** [comment]
- checkout the 1.3 tag
- alter build so it added appropriate Automatic-Module-Name to each jar generated
- optionally alter build so if any dependencies/plugins have updates, change the build to use them
- publish those jar's/pom's to maven central

For more information about how to determine what module name to use please refer to this site http://blog.joda.org/2017/04/java-se-9-jpms-module-naming.html

**Comment by nhojpatrick** [comment]
@sf105 got delayed but finally got time to get a 1.3.1 proposed release.

https://github.com/nhojpatrick/JavaHamcrest/tree/bugfix/204_1.3.x_jpms

All it does it add a Automatic-Module-Name into the MANIFEST.MF.

This is purely so those depending upon hamcrest don't need to use jar name generated module names. i.e. hamcrest-alpha-bravo-1.2.3.jar would become module name hamcrest.alpha.bravo

The current BUILDING.TXT talks about Java 1.5 and Ant 1.6. The oldest Java I've got access to is Java 1.7.0_80 and I downloaded Ant 1.6.5. But the tests with the following but I couldn't work out how to fix that quickly.

```
    [junit] Testcase: warning(junit.framework.TestSuite$1):	FAILED
    [junit] No tests found in org.hamcrest.core.CombinableTest
    [junit] junit.framework.AssertionFailedError: No tests found in org.hamcrest.core.CombinableTest
```

Don't merge this branch into master, I would create hamcrest-java-1.3.1 branch and tag, the do `ant clean all -Dversion=1.3.1` then upload into maven central.

If you happy with the proposed module names and format, I'll start looking into sorting out similar for master branch.

**Comment by sf105** [comment]
The building.txt is out of date as we moved to Groovy (see another issue about moving back to maven). I start working on updating it to a more recent version that actually support uploading to maven central, but lost steam in the process.

**Comment by sf105** [comment]
I've made you a committer if you want to start work on this.

**Comment by xenoterracide** [comment]
found this looking at a problem which just turned out to be a wrong requires name... does this still need to be open? looks like this is all in Manifest.MF now.

It would be useful to have it in a real module-info.java for jlink if you're depending on java 11+ now (I think that it can also be put into a `java9` directory but I don't know exactly how that needs to be set up, and whether it works with jlink.

note: it's `org.hamcrest` if anyone gets their IDE trying `hamcrest.core`

**Comment by tumbarumba** [comment]
Thanks for the prompt @xenoterracide. This is a redundant issue, as it was addressed in a different issue and PR (see #185  and #186). I'm closing this issue now, although we may revisit this in future as we start to natively support more recent versions of Java
